### PR TITLE
GUVNOR-3112: Package list box should initialize with "groupId.artifactId" package selected (considering reserved words)

### PR DESCRIPTION
**Pull request by paulovmr**

### CODE REVIEW

File: guvnor-project/guvnor-project-api/src/main/java/org/guvnor/common/services/project/service/ProjectResourceResolver.java [hunk 1]
```diff
@@ -28,42 +28,44 @@
...
-    Set<Package> resolvePackages( final Package pkg );
+    Set<Package> resolvePackages(final Package pkg);
 
-    Package resolveDefaultPackage( final Project project );
+    Package resolveDefaultPackage(final Project project);
 
-    Package resolveParentPackage( final Package pkg );
+    Package resolveDefaultWorkspacePackage(final Project project);
+
+    Package resolveParentPackage(final Package pkg);
```
Comment (paulovmr) [inline guvnor-project/guvnor-project-api/src/main/java/org/guvnor/common/services/project/service/ProjectResourceResolver.java]: Real change on this file is to add this method.

File: guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ResourceResolver.java [hunk 2]
```diff
@@ -331,41 +336,52 @@ public Project resolveToParentProject( final Path resource ) {
     }
 
     @Override
-    public Package resolveParentPackage( final Package pkg ) {
+    public Package resolveDefaultWorkspacePackage(final Project project) {
```
Comment (paulovmr) [inline guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ResourceResolver.java]: Real change.

File: guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ResourceResolver.java [hunk 3]
```diff
@@ -374,229 +390,240 @@ public Package resolveParentPackage( final Package pkg ) {
...
+                               resource);
+        } catch (Exception e) {
+            throw ExceptionUtilities.handleException(e);
         }
     }
 
-    protected boolean hasPom( final org.uberfire.java.nio.file.Path path ) {
-        final org.uberfire.java.nio.file.Path pomPath = path.resolve( POM_PATH );
-        return Files.exists( pomPath );
+    public String getDefaultWorkspacePath(final GAV gav) {
```
Comment (paulovmr) [inline guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ResourceResolver.java]: Real change.

File: guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ResourceResolver.java [hunk 4]
```diff
@@ -374,229 +390,240 @@ public Package resolveParentPackage( final Package pkg ) {
...
+        return StringUtils.join(getLegalId(gav.getGroupId()),
+                                "/") + "/" + StringUtils.join(getLegalId(gav.getArtifactId()),
+                                                              "/");
     }
 
-    protected T makeProject( final org.uberfire.java.nio.file.Path nioProjectRootPath ) {
-        final T project = simpleProjectInstance( nioProjectRootPath );
+    public String[] getLegalId(final String id) {
```
Comment (paulovmr) [inline guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ResourceResolver.java]: Real change.

**Comment by paulovmr** [comment]
@manstis Hi Mike, if you could take a look, please :)

File: guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ResourceResolver.java (hunk 2 above)
Comment (jhrcek) [inline guvnor-project/guvnor-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/ResourceResolver.java]: @paulovmr @manstis how about formatting the whole project in separate PR using the agreed upon formatting conventions? This way we could avoid having to look for "real" changes in the see of formatting changes.. ;-)

**Pull request merged**